File: solutions/0001. Two Sum/0001.js
// Time complexity: O(n)
// Space complexity: O(n)
var twoSum = function(nums, target) {
    const numToIndex = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (numToIndex.has(target - nums[i])) {
            return [numToIndex.get(target - nums[i]), i];
        }
        numToIndex.set(nums[i], i);
    }
    return [-1, -1];
};

File: solutions/0001. Two Sum/0001.cs
// Time complexity: O(n)
// Space complexity: O(n)
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numToIndex = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; ++i) {
            if (numToIndex.ContainsKey(target - nums[i])) {
                return new int[] { numToIndex[target - nums[i]], i };
            }
            numToIndex[nums[i]] = i;
        }
        throw new Exception("No two numbers sum to the target.");
    }
}